app/src/test/kotlin/io/github/seggan/myxal/TestHelper.kt

```diff
@@ -2,30 +2,10 @@ package io.github.seggan.myxal
 
 import io.github.seggan.myxal.app.Main.doMain
 import java.io.IOException
-import java.net.URLClassLoader
-import java.nio.file.Files
-import java.nio.file.Path
-import java.util.regex.Pattern
 
 object TestHelper {
-    private val p = Pattern.compile("\\..*$")
-
     @Throws(IOException::class)
     fun run(vararg args: String) {
-        val fileName = Path.of(args[0]).fileName.toString()
-        doMain(arrayOf(*args), true)
-        val jar = Path.of(args[0]).resolveSibling(p.matcher(fileName).replaceAll(".jar"))
-        val cl = URLClassLoader(
-            arrayOf(jar.toUri().toURL()),
-            TestHelper::class.java.classLoader
-        )
-        try {
-            val clazz = cl.loadClass("jyxal.Main")
-            clazz.getMethod("main", Array<String>::class.java).invoke(null, arrayOfNulls<String>(0) as Any)
-        } catch (e: ReflectiveOperationException) {
-            throw RuntimeException(e)
-        }
-        cl.close()
-        Files.delete(jar)
+        doMain(arrayOf(".", "-r", *args), true)
```

Comment:
Well, actually, the test harness can be simplified now since the runtime is located in the full package `io....` instead of `runtime....`. So we can simply use a dynamic class loader that loads the bytecode of `myxal.Main` into memory and directly executes it for testing. But ig ill do that myself.

Reply:
So I just did that, but I'm not sure how to apply Proguard optimizations if you're loading directly